### PR TITLE
New system test workflow automation and inclusion in suite

### PR DESCRIPTION
**Pull request by sumabai**
# Description
Jira: https://issues.redhat.com/secure/RapidBoard.jspa?rapidView=12700&view=detail&selectedIssue=RHCEPHQE-16989&quickFilter=113035#

Fix Description: Automated SV and Clone bulk create and delete ops and included it into system suite to run in parallel with Client IO workflows.

Logs:
Execution logs for SV creates and deletes: 
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-KA0E8A 
Note:
By default, not able to run even 100 parallel creates and deletes too, may be number of max ssh connections are limited. This issue is seen when when workflow was run in silos, but once automation is finalised, it would be run in parallel to many other workflows, so having more ssh sessions to handle parallelism.
Since Baremetal cluster is not healthy, happened to run on VM config. Could be a ssh session limit on VM? For now will manage with 100 parallel creates and deletes in once batch and reach till 1k subvols.
Once baremetal execution is complete, will alter batch size.

Execution logs for Clone bulk create and delete workflow:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-GUDGJ3  
Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_system/cephfs_systest_sv_clone_ops.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+logging_thread = None
+mds_logging_thread = None
+stop_event = threading.Event()
+
+
+def run(ceph_cluster, **kw):
+    """
+    System test - SV and Clone ops : Run below workflows in parallel and repeat until runtime,
+
+    1. Create 100 subvolumes. Delete the subvolumes and in parallel create another 100 subvolumes.
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_system/cephfs_systest_sv_clone_ops.py:30]: It is good to have these configurations made custom variables and pass to the test. If next time, scaling is needed, we could increase the count without modifying the code.

File: tests/cephfs/cephfs_system/cephfs_systest_sv_clone_ops.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+    out, _ = client.exec_command(sudo=True, cmd=cmd)
+    output = json.loads(out)
+    group_name = output[0]["name"]
+    sv_rm_list = []
+    for k in range(0, sv_cnt):
+        sv_name = f"systest_sv_{k}"
+        sv_obj = {
+            "sv_name": sv_name,
+        }
+        cmd = f"ceph fs subvolume create {fs_name} {sv_name} {group_name}"
```
Comment (AmarnatReddy) [inline tests/cephfs/cephfs_system/cephfs_systest_sv_clone_ops.py:183]: For subvolume Operations can we use common functions which we have in Utils file . if there are any changes we can address at single point. 
Same for clone operations as well

File: tests/cephfs/cephfs_system/cephfs_systest_sv_clone_ops.py (hunk 1 above)
Comment (sumabai) [reply tests/cephfs/cephfs_system/cephfs_systest_sv_clone_ops.py:30]: It is customizable neha, please check code below at line40,
        sv_cnt = config.get("sv_cnt", 300)
        clone_cnt = config.get("clone_cnt", 10)

File: tests/cephfs/cephfs_system/cephfs_systest_sv_clone_ops.py (hunk 2 above)
Comment (sumabai) [reply tests/cephfs/cephfs_system/cephfs_systest_sv_clone_ops.py:183]: The intent here is to invoke create and delete commands in parallel. Relying on test_libs for these may add delays due to additional implementations in testlib module like param validations, hence decided to run them instantly.
Will try using lib modules instead and see if parallelism can still be achieved.

**Review by hkadam134 — APPROVED** [review]

File: tests/cephfs/cephfs_system/cephfs_systest_sv_clone_ops.py (hunk 2 above)
Comment (neha-gangadhar) [reply tests/cephfs/cephfs_system/cephfs_systest_sv_clone_ops.py:183]: +1 to reusing common functions. The intent of adding common functions in libraries is for reuse. Encourage to use them going forward.

**Review by neha-gangadhar — APPROVED** [review]
Approved

**Pull request merged**